index.bs

```diff
@@ -819,76 +860,66 @@ spec:url; type:dfn; text:scheme
   typedef (WindowProxy or MessagePort or ServiceWorker or HTMLPortalElement or PortalHost) MessageEventSource;
   </xmp>
 
-  A {{PortalHost}} is exposed at times when the window may be in a [=portal browsing context=].
+  Event handlers {#patch-event-handlers}
+  --------------------------------------
 
-  <xmp class="idl">
-      partial interface Window {
-          readonly attribute PortalHost? portalHost;
-      };
+  The table of [=event handlers=] which must be supported by {{Window}} objects, as [=event handler
+  IDL attributes=] on the {{Window}} objects themselves (i.e. the table containing `onafterprint`),
+  gets extended with the following row:
+
+  <table class="data" dfn-for="Window">
+    <thead>
+      <tr>
+        <th>[=Event handler=]</th>
+        <th>[=Event handler event type=]</th>
+      </tr>
+    </thead>
+    <tbody>
+      <tr>
+        <td><dfn attribute for="Window"><code>onportalactivate</code></dfn></td>
+        <td>{{portalactivate}}</td>
+      </tr>
+    </tbody>
+  </table>
 
+  The corresponding {{WindowEventHandlers}} mixin gets extended as follows:
+
+  <xmp class="idl">
       partial interface mixin WindowEventHandlers {
           attribute EventHandler onportalactivate;
       };
   </xmp>
 
-  <section algorithm="window-portalhost">
-    The <dfn attribute for="Window">portalHost</dfn> attribute's getter *must* run the following steps:
-
-    1. Let |context| be [=this=]'s [=Window/browsing context=].
-
-    1. If |context| is null or the [=portal state=] of |context| is not "`portal`", then return null.
-
-    1. Return [=this=]'s [=portal host object=].
-  </section>
-
-  <wpt>
-    portals-host-exposure.sub.html
-    portals-host-null.html
-  </wpt>
-
-  The following events are dispatched on {{Window}} objects:
+  The <a href="https://html.spec.whatwg.org/#events-2">Events index</a> is also updated with the
+  following additional row:
 
   <table class="data" dfn-for="Window">
     <thead>
       <tr>
-        <th>Event name</th>
+        <th>Event</th>
         <th>Interface</th>
-        <th>Dispatched when</th>
+        <th>Interesting targets</th>
+        <th>Description</th>
       </tr>
     </thead>
     <tbody>
       <tr>
         <td><dfn event for="Window"><code>portalactivate</code></dfn></td>
         <td>{{PortalActivateEvent}}</td>
-        <td>The window is associated with a new [=top-level browsing context=] due to activation of its [=portal browsing context=].</td>
+        <td>{{Window}}</td>
+        <td>Fired at the {{Window}} when it is associated with a new [=top-level browsing context=]
```

Comment:
Well, the {{Window}} (i.e. the object itself) is still associated with the same top-level browsing context; it's the UA-level window that is associated with a new browsing context. This wording was awkward before this PR too (probably because this was one of the earlier things to be written and terminology has changed).

Reply:
Hmm, yeah, I see. Maybe something like this?

> Fired at the Window of a portal browsing context when that portal browsing context is activated.

Reply:
Sure, sg. The exact details are elsewhere, so this only needs to give the right general intution.